chainbase/src/main/java/org/tron/core/service/RewardViCalService.java

```diff
@@ -75,8 +74,14 @@ public RewardViCalService(@Autowired  DynamicPropertiesStore propertiesStore,
     this.witnessStore = witnessStore.getDb();
   }
 
-  @PostConstruct
-  private void init() {
+  public void init() {
+    // after init, we can get the latest block header number from db
+    this.newRewardCalStartCycle = this.getNewRewardAlgorithmEffectiveCycle();
+    boolean ret = this.newRewardCalStartCycle != Long.MAX_VALUE;
+    if (ret) {
+      // checkpoint is flushed to db, we can start rewardViCalService immediately
+      lastBlockNumber = Long.MAX_VALUE;
+    }
     es.scheduleWithFixedDelay(this::maybeRun, 0, 3, TimeUnit.SECONDS);
```

Comment:
What is the significance of this code in the enableNewRewardAlgorithm function? 

if (ret && lastBlockNumber == -1) {
      lastBlockNumber = this.getLatestBlockHeaderNumber();
    }

Reply:
For the block-catching case, record the block for which NewRewardAlgorithm is in effect, and wait for the block to be flushed before data computation to ensure data consistency.

Reply:
This function getLatestBlockHeaderNumber obtains the data of the underlying database, right?

Reply:
Yes.